lib/logidze/meta.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+module Logidze # :nodoc:
+  # Provide methods to attach meta information
+  module Meta
+    def with_meta(meta, &block)
+      MetaTransaction.wrap_with(meta, &block)
+    end
+
+    def with_responsible(responsible_id, &block)
+      return yield if responsible_id.nil?
+      meta = { Logidze::History::Version::META_RESPONSIBLE => responsible_id }
+      with_meta(meta, &block)
+    end
+
+    class MetaTransaction # :nodoc:
+      def self.wrap_with(meta, &block)
+        new(meta, &block).perform
+      end
+
+      attr_reader :meta, :block
+
+      delegate :connection, to: ActiveRecord::Base
+
+      def initialize(meta, &block)
+        @meta = meta
+        @block = block
+      end
+
+      def perform
+        return if block.nil?
+        return block.call if meta.nil?
+        ActiveRecord::Base.transaction { call_block_in_meta_context }
+      end
+
+      private
+
+      def call_block_in_meta_context
+        prev_meta = current_meta
+
+        meta_stack.push(meta)
+
+        pg_set_meta_param(current_meta)
+        result = block.call
+        pg_reset_meta_param(prev_meta)
+
+        result
+      ensure
```

Comment:
I don't see whether we have a spec for this case? We need to make sure that we pop from `meta_stack` when we're leaving `with_meta` block

Reply:
Added couple of checks to the existing specs and one separate spec to handle nested blocks. I wonder if there is a better way to test the same thing without letting the test know that we use `Thread.current`, maybe it worth trying to update `subject` in different places and verify that it's meta is correct